--- ./src/bin/pg_upgrade/option.c
+++ option.c
@@ -94,9 +94,11 @@
 		}
 	}
 
+#ifndef SNAPPY
 	/* Allow help and version to be run as root, so do the test here. */
 	if (os_user_effective_id == 0)
 		pg_fatal("%s: cannot be run as root\n", os_info.progname);
+#endif
 
 	if ((log_opts.internal = fopen_priv(INTERNAL_LOG_FILE, "a")) == NULL)
 		pg_fatal("cannot write to log file %s\n", INTERNAL_LOG_FILE);
